src/app/core/entity/entity-actions/entity-actions.service.spec.ts

```diff
@@ -80,19 +92,42 @@ describe("EntityActionsService", () => {
     expect(mockedEntityDeleteService.deleteEntity).not.toHaveBeenCalled();
   });
 
-  it("should delete entity, show snackbar confirmation and navigate back", async () => {
+  it("should delete a single entity, show snackbar confirmation and navigate back", async () => {
     // onAction is never called
     mockSnackBarRef.onAction.and.returnValues(NEVER);
     mockSnackBarRef.afterDismissed.and.returnValue(of(undefined));
 
-    const result = await service.delete(new Entity(), true);
+    const result = await service.delete(primaryEntity, true);
 
     expect(result).toBe(true);
     expect(snackBarSpy.open).toHaveBeenCalled();
-    expect(mockedEntityDeleteService.deleteEntity).toHaveBeenCalled();
+    expect(mockedEntityDeleteService.deleteEntity).toHaveBeenCalledWith(
+      primaryEntity,
+    );
     expect(mockRouter.navigate).toHaveBeenCalled();
   });
 
+  it("should delete several entities and show snackbar confirmation", async () => {
```

Comment:
We should add a test to confirm that the undo function actually includes all the original entities if multiple were acted on

Reply:
I added such tests for the deletion and anonymization of several entities.